src/plugin/luapi_application.h

```diff
@@ -475,6 +476,194 @@ static int applib_changeBackgroundPdfPageNr(lua_State* L) {
+
+    std::string size = toolSizeToString(toolHandler->getPenSize());
+    int fillOpacity = toolHandler->getPenFill();
+    bool fill = toolHandler->getPenFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the highlighter in a Lua table of the shape
+ * {
+ *   "size" = string
+ *   "fill" = bool
+ *   "fillOpacity" = integer
+ * }
+ *
+ * See /src/control/ToolEnums.cpp for possible values of "size".
+ *
+ * Example 1: local highlighterInfo = app.getHighlighterInfo()
+ *            local size = highlighterInfo["size"]
+ *            local opacity = highlighterInfo["fillOpacity"]
+ *
+ * Example 2: local fill = app.getHighlighterInfo()["fill"]
+ */
+static int applib_getHighlighterInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string size = toolSizeToString(toolHandler->getHighlighterSize());
+    int fillOpacity = toolHandler->getHighlighterFill();
+    bool fill = toolHandler->getHighlighterFillEnabled();
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fill");
+    lua_pushboolean(L, fill);
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "fillOpacity");
+    lua_pushinteger(L, fillOpacity);
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+/*
+ * Returns a table encoding all info on the eraser in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
+ * }
+ *
+ * See /src/control/ToolEnums.cpp for possible values of "type" and "size".
+ *
+ * Example 1: local eraserInfo = app.getEraserInfo()
+ *            local type = eraserInfo["type"]
+ *            local size = eraserInfo["size"]
+ *
+ * Example 2: local type = app.getEraserInfo()["type"]
+ */
+static int applib_getEraserInfo(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* control = plugin->getControl();
+    ToolHandler* toolHandler = control->getToolHandler();
+
+    std::string type = eraserTypeToString(toolHandler->getEraserType());
+    std::string size = toolSizeToString(toolHandler->getEraserSize());
+
+    lua_newtable(L);
+
+    lua_pushliteral(L, "type");
+    lua_pushstring(L, type.c_str());
+    lua_settable(L, -3);
+
+    lua_pushliteral(L, "size");
+    lua_pushstring(L, size.c_str());
+    lua_settable(L, -3);
+
+    return 1;
+}
+
+
+/*
+ * Returns a table encoding all info on the active tool in a Lua table of the shape
+ * {
+ *   "type" = string
+ *   "size" = string
+ *   "color" = integer
```

Comment:
It might make sense to break `color` up into `red`, `green`, and `blue` components. Alternatively, document how to access the different components of `color`.

Reply:
I think we should remain consistent with how we pass colors in `app.changeToolColor`
```lua
app.changeToolColor({["color"] = 0xff00ff, ["tool"] = "PEN"})
```
From the current implementation one can get the string representation via 
```lua
local color = app.getActiveToolInfo()["color"]
local hexString = string.format("0x%06x", color)
```
and from there one can extract the r,g,b values (between 0 and 255) via
```lua
red = tonumber("0x" .. string.sub(hexString,3,4))
green = tonumber("0x" .. string.sub(hexString,5,6))
blue = tonumber("0x" .. string.sub(hexString,7,8))
```

So I think we should just add it to the documentation (in one of the examples)

Reply:
Or more directly:
```lua
local color = app.getActiveToolInfo()["color"]
local red = color >> 16 & 0xff
local green = color >> 8 & 0xff
local blue = color & 0xff
```

Reply:
Okay. That makes sense.

It might also be worth [creating a `Color` type for `lua`](https://www.lua.org/pil/28.html) with `__tonumber` and `getR()`, `getG()`, and `getB()` methods. In this case, `changeToolColor` could accept an `integer` or a `xournalpp.Color`.

The benefits of a custom type:
 * Allows us to add additional methods to `Color` in the future (e.g. hueRotate, invert, etc).
 * Easier to get/set `RGB` components.

Some downsides:
 * More complicated to implement.
   * I don't think we're defining custom types via the plugin API yet...
 * Need to update `app.changeToolColor`

Reply:
Overall, I think using a number is fine here. Let's just make sure to document it on the website :).